src/main/java/com/redhat/lightblue/healthcheck/HealthCheckResource.java

```diff
@@ -75,41 +131,41 @@ public Response health() {
             updateRequest.where(Query.withValue("_id", Query.eq, uuid));
             updateRequest.returns(Projection.excludeFieldRecursively("*"));
             updateRequest.updates(Update.set("value", "updated"));
-            client.data(updateRequest);
+            client.getValue().data(updateRequest);
 
-            found = find(uuid);
+            found = find(client.getValue(), uuid);
 
             assertNotNull(found);
             assertEquals(hostname, found.getHostname());
             assertEquals("updated", found.getValue());
 
             DataDeleteRequest deleteRequest = new DataDeleteRequest(ENTITY);
             deleteRequest.where(
-                Query.or(
-                    Query.withValue("_id", Query.eq, uuid),
-                    Query.withValue("creationDate", Query.lte,
-                            Date.from(LocalDateTime
-                                    .now()
-                                    .minusMinutes(5)
-                                    .atZone(ZoneId.systemDefault())
-                                    .toInstant()))
-                )
+                    Query.or(
+                            Query.withValue("_id", Query.eq, uuid),
+                            Query.withValue("creationDate", Query.lte,
+                                    Date.from(LocalDateTime
+                                            .now()
+                                            .minusMinutes(5)
+                                            .atZone(ZoneId.systemDefault())
+                                            .toInstant()))
+                    )
             );
-            client.data(deleteRequest);
+            client.getValue().data(deleteRequest);
 
-            assertNull(find(uuid));
+            assertNull(find(client.getValue(), uuid));
 
             LOGGER.debug("Health check passed.");
-            return Response.status(Status.OK).entity(
-                    "{\"status\":\"success\"}").build();
+
+            return new HealthCheckStatus(HealthCheckStatus.Status.success);
         } catch (Throwable e) {
             LOGGER.error("Health check failed.", e);
-            return Response.status(Status.INTERNAL_SERVER_ERROR).entity(
-                    "{\"status\":\"error\",\"message\":\"" + e.getMessage() + "\"}").build();
+            String errorMessage = " [client=" + client.getKey() + " error=" + e.getMessage() + "] ";
+            return new HealthCheckStatus(HealthCheckStatus.Status.error).withErrorMessage(errorMessage);
```

Comment:
I feel like maybe this reads kind of funny. Maybe instead of an enum, HealthCheckStatus had factory methods on it, one for success and another for failure?